gateway-service/src/main/java/org/zowe/apiml/gateway/filters/security/BasicAuthFilter.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.gateway.filters.security;
+
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.security.core.context.ReactiveSecurityContextHolder;
+import org.springframework.web.server.ServerWebExchange;
+import org.springframework.web.server.WebFilter;
+import org.springframework.web.server.WebFilterChain;
+import org.zowe.apiml.gateway.service.BasicAuthProvider;
+import reactor.core.publisher.Mono;
+
+import java.util.Optional;
+
+@RequiredArgsConstructor
+public class BasicAuthFilter implements WebFilter {
+
+    private final BasicAuthProvider basicAuthProvider;
+
+    @Override
+    public Mono<Void> filter(ServerWebExchange exchange, WebFilterChain chain) {
+        var authHeader = resolveAuth(exchange.getRequest());
+        return authHeader.map(header -> basicAuthProvider.getToken(header).flatMap(token -> {
```

Comment:
Does it make sense to send a cookie with the token to the user for the next request?

Reply:
Why does the basic auth filter take care of a token?

Reply:
Yes, because it is about the response from ZAAS.